plugins/utils/hardtime.nix

```diff
@@ -0,0 +1,169 @@
+{
+  config,
+  pkgs,
+  lib,
+  ...
+}:
+with lib; let
+  cfg = config.plugins.hardtime;
+  helpers = import ../helpers.nix {inherit lib;};
+in {
+  options = {
+    plugins.hardtime =
+      helpers.extraOptionsOptions
+      // {
+        enable = mkEnableOption "Enable hardtime.";
+        package = helpers.mkPackageOption "hardtime" pkgs.vimPlugins.hardtime-nvim;
+
+        maxTime = helpers.defaultNullOpts.mkInt 1000 ''
+          Maximum time (in milliseconds) to consider key presses as repeated.
+        '';
+
+        maxCount = helpers.defaultNullOpts.mkInt 2 ''
+          Maximum count of repeated key presses allowed within the `max_time` period.
+        '';
```

Comment:
For those you may want to use `mkUnsignedInt` (>=0) or `mkPositiveInt (>0) instead.